packages/@react-aria/tooltip/src/useTooltip.ts

```diff
@@ -10,45 +10,29 @@
  * governing permissions and limitations under the License.
  */
 
-import {DOMProps} from '@react-types/shared';
 import {DOMPropsResponderContext} from '@react-aria/interactions';
 import {HTMLAttributes, useContext} from 'react';
+import {SpectrumTooltipProps} from '@react-types/tooltip';
 import {useId} from '@react-aria/utils';
 
-interface TooltipProps extends DOMProps {
-  role?: 'tooltip'
-  id?: string
-}
-
 interface TooltipAria {
   tooltipProps: HTMLAttributes<HTMLElement>
 }
 
-export function useTooltip(props: TooltipProps): TooltipAria {
-  let contextProps = useContext(DOMPropsResponderContext);
-  let tooltipId = useId(props.id);
-
+export function useTooltip(props: SpectrumTooltipProps): TooltipAria {
   let {
+    id,
     role = 'tooltip'
   } = props;
 
-  let tooltipProps: TooltipAria['tooltipProps'] = {
-    role,
-    id: tooltipId
-  };
-
-  if (contextProps) {
-    if (contextProps.onPointerLeave && contextProps.onPointerEnter) {
-      tooltipProps.onPointerLeave = contextProps.onPointerLeave;
-      tooltipProps.onPointerEnter = contextProps.onPointerEnter;
-    }
-    if (contextProps.onMouseLeave && contextProps.onMouseEnter) {
-      tooltipProps.onMouseLeave = contextProps.onMouseLeave;
-      tooltipProps.onMouseEnter = contextProps.onMouseEnter;
-    }
-  }
+  let {onPointerLeave, onPointerEnter} = useContext(DOMPropsResponderContext) || {};
```

Comment:
I would make sure to pass all props through here, not just these two. There will be more in other cases, like `onMouseLeave`/`onMouseEnter`, etc. Best to just pass them all through.